apisix/plugins/proxy-rewrite.lua

```diff
@@ -153,10 +154,20 @@ function _M.rewrite(conf, ctx)
         end
     end
 
-    upstream_uri = core.utils.uri_safe_encode(upstream_uri)
+    local index, _ = find_str(upstream_uri, "?")
+    if index then
+        upstream_uri = core.utils.uri_safe_encode(sub_str(upstream_uri, 1, index-1)) ..
```

Comment:
missing `?`

Reply:
sub_str(upstream_uri, index) start with `?`

Reply:
> missing `?`

sub_str(upstream_uri, index) start with `?`

Reply:
I think your way is correct. Ignore what I said.